### ISSUE TITLE
componentWillReceiveProps calls "initialize" without fields paramter

### ISSUE THREAD

**Issue by vladimirbuskin** [erikras/redux-form#437]
in my case after form is submited, object update in state and that makes 
initialValues to change to changed object

**createHigherOrderComponent.js**
```javascript
      ReduxForm.prototype.componentWillReceiveProps = function componentWillReceiveProps(nextProps) {
        if (!_deepEqual2['default'](this.props.fields, nextProps.fields) || !_deepEqual2['default'](this.props.form, nextProps.form)) {
          this.fields = _readFields2['default'](nextProps, this.fields, this.asyncValidate, isReactNative);
        }
        if (!_deepEqual2['default'](this.props.initialValues, nextProps.initialValues)) {
          this.props.initialize(nextProps.initialValues  /* here */);
        }
      };
```

after i change that to 

```javascript
      ReduxForm.prototype.componentWillReceiveProps = function componentWillReceiveProps(nextProps) {
        if (!_deepEqual2['default'](this.props.fields, nextProps.fields) || !_deepEqual2['default'](this.props.form, nextProps.form)) {
          this.fields = _readFields2['default'](nextProps, this.fields, this.asyncValidate, isReactNative);
        }
        if (!_deepEqual2['default'](this.props.initialValues, nextProps.initialValues)) {
          this.props.initialize(nextProps.initialValues, nextProps.fields);
        }
      };
```

everything seems to work fine for me.

**Comment by vladimirbuskin** [comment]
my Form source if that helps
```javascript
import React from 'react';
import { Link } from 'react-router';
import { reduxForm } from 'redux-form';
import validation from './dashboardEditValidation';
import { MSDropdown, MSInput } from '../components/'
import { Button } from '../bootstrap';
import { editDashboard } from '../actions/adminDashboardActions';


class DashboardEdit extends React.Component {

  constructor(props) {
    super(props);

    this.onSubmit = this.onSubmit.bind(this);
  }

  getBackUrl() {
    return `/profile/${this.props.params.profile}/dashboard/`;
  }

  onSubmit(data) {

    const {
      params: {
        profile,
        id
      },
      dashboardList
    } = this.props;

    var dashboard = dashboardList.find(x=>x.id==id*1);

    const res = {
      ...dashboard,
      ...data,
      profile,
      id:id*1,
      profileId: 2,
      entityId: Math.random().toString(36).substring(7),
    };

    this.props.editDashboard(profile, res, this.getBackUrl());
  }

  render() {
    const {
      fields: { name, datasetId },
      datasetList,
      handleSubmit,
      initializeForm
    } = this.props;

    return (
      <div className="container">
        <h1>Add Dashboard</h1>

        <Link to={this.getBackUrl()}>
          <Button><i className="fa fa-chevron-left"></i> Back</Button>
        </Link>

        <div className="row">
          <div className="col-xs-6">
            <form style={{marginTop:20}} onSubmit={handleSubmit(this.onSubmit)}>
              <div className='form-group'>
                <MSInput type='text'
                         label='Name'
                         className='form-control'
                  {...name} />
              </div>
              <div className='form-group'>
                <MSDropdown label="Dataset" valueField='id'
                            textField='name'
                            data={datasetList}
                  {...datasetId} />
              </div>
              <div className='form-group'>
                <button className='btn btn-primary btn-lg'>Submit</button>
                &nbsp;&nbsp;
                <button className='btn btn-default btn-lg' onClick={initializeForm}>Clear</button>
              </div>
            </form>
          </div>
        </div>
      </div>
    )
  }
}

export default reduxForm(
  // form mapping
  {
    form: 'dashboard',
    fields: ['name','datasetId'],
    validate: validation,
    destroyOnUnmount: false
  },
  // state mapping
  state => ({
    user: state.auth.user,
    datasetList: state.profile.datasetList,
    dashboardList: state.profile.dashboardList,
    initialValues: state.routing.params.id ? state.profile.dashboardList.find(x=>x.id === state.routing.params.id*1) : {}
  }),
  // functions mapping
  {
    editDashboard
  }
)(DashboardEdit)
```

**Comment by latentflip** [comment]
@vladimirbuskin this was fixed in 4.0.1: https://github.com/erikras/redux-form/commit/1e6d36300afc7a0d41b7d125b21df801b8978a62 if you update you should be good!
